codegen/src/next/mod.rs

```diff
@@ -339,46 +416,89 @@ mod tests {
         };
 
         let actual = quote!(#endpoint);
-        let expected = quote!(.route("/hello", shuttle_next::routing::get(hello)));
```

Comment:
Won't this error on `get` being unknown?

Reply:
The reason I removed the full paths is it was causing a bug in chaining the handlers when running the example. In the tests it worked and gave the correct output: 
```rust
let mut router = shuttle_next::Router::new()
        .route("/goodbye", shuttle_next::routing::get(get_goodbye).shuttle_next::routing::post(post_goodbye))
        .route("/hello", shuttle_next::routing::get(hello));
```
But when compiling the example it generated this:
```rust
let mut router = shuttle_next::Router::new()
        .route(
            "/goodbye",
            shuttle_next::routing::get(get_goodbye).shuttle_next,
            ::routing::post(post_goodbye),
        )
        .route("/hello", shuttle_next::routing::get(hello))
```

When removing the full path it works, and I make sure it's imported by adding it to the `app` in the `App` `ToTokens` impl with  `use shuttle_next::routing::*`. I am not very happy with that solution though, and the bug doesn't make sense to me, so I'm sure I've done something wrong here.

Reply:
Ah, I see. I think the use will be fine :smile: 

Fyi, this will be the line that you will need to edit correctly to not have the `use`
https://github.com/shuttle-hq/shuttle/pull/588/files#diff-ef21a60b2fde7fdab29d9510e71bb56dc2c9f4e85f06bf0a4b38a8c1ceef00e1R247

Reply:
Yeah, I tried it like this: 
```rust
let route = quote!(.route(#route, #(shuttle_next::routing::#handlers).*));
```
I still get the same bug, but I'm pretty sure this is correct

Reply:
It's almost correct - you want the namespace outside the repeat:

```rust
let route = quote!(.route(#route, shuttle_next::routing::#(#handlers).*));
```

Reply:
Ah, of course :facepalm: Thanks, that works!